src/pkg/cli/compose/convert.go

```diff
@@ -13,6 +13,33 @@ import (
 	compose "github.com/compose-spec/compose-go/v2/types"
 )
 
+type serviceNameReplacer struct {
+	client                     client.Client
+	nonReplaceServiceNameRegex *regexp.Regexp
+	serviceName                string
+	serviceNameRegex           *regexp.Regexp
+}
+
+func (s *serviceNameReplacer) replaceServiceNameWithDNS(key, value string, showWarnings bool) string {
+	val := value
+	if s.serviceNameRegex != nil {
+		// Replace service names with their actual DNS names; TODO: support public names too
+		val = s.serviceNameRegex.ReplaceAllStringFunc(value, func(serviceName string) string {
+			return s.client.ServiceDNS(NormalizeServiceName(serviceName))
+		})
+
+		if showWarnings {
+			if val != value {
+				term.Warnf("service %q: service name was fixed up: environment variable %q assigned value %q", s.serviceName, key, val)
```

Comment:
These warnings refer to "environment variable" but we use this function for "build args" too now, so misleading.